src/message.ts

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+import get from 'lodash.get';
+import { CumulusMessage } from '@cumulus/types/message';
+import { cumulusMessageWithGranulesInPayload } from './types';
+
+const GRANULE_LOG_LIMIT = 500;
+
+/**
+ * Get granules from execution message.
+ *   Uses the order of precedence as defined by the cumulus/common/message
+ *   description.
+ *
+ * @param {Object} message - a Cumulus message
+ * @param {integer} granuleLimit - number of granules to limit the log to
+ * including, to avoid environment variable truncation
+ * @returns {Array<Object>} - An array of granule ids
+ */
+export const getMessageGranules = (
+  message: cumulusMessageWithGranulesInPayload,
+  granuleLimit: number = GRANULE_LOG_LIMIT
+) => {
+  const granules = message?.payload?.granules
+        || get(message, 'meta.input_granules')
```

Comment:
Do we need get here, or could we use `?.`

Reply:
We don't, and also we don't need to account for cma.event type events as they are never evaluated : 420deba 529fc78

and furthermore we don't need lodash.get: 8dac78e.